src/iosrv/tests/drpc_client_tests.c

```diff
@@ -235,6 +240,296 @@ test_drpc_verify_notify_bio_error(void **state)
+	assert_non_null(svc_ranks);
+
+	assert_int_equal(ds_notify_pool_svc_update(NULL, svc_ranks),
+			 -DER_INVAL);
+	verify_notify_pool_svc_update_inval();
+
+	d_rank_list_free(svc_ranks);
+
+	drpc_fini();
+}
+
+static void
+verify_notify_ras(ras_event_t id, char *msg, ras_type_t type,
+		  ras_sev_t sev, char *hid, d_rank_t *rank, char *jid,
+		  uuid_t *puuid, uuid_t *cuuid, daos_obj_id_t *oid,
+		  char *cop, char *data)
+{
+	Drpc__Call		*call;
+	Shared__ClusterEventReq	*req;
+	uuid_t			 req_puuid, req_cuuid;
+
+	call = drpc__call__unpack(NULL, sendmsg_msg_iov_len,
+				  sendmsg_msg_content);
+	assert_non_null(call);
+	assert_int_equal(call->module, DRPC_MODULE_SRV);
+	assert_int_equal(call->method, DRPC_METHOD_SRV_CLUSTER_EVENT);
+
+	/* Verify payload contents */
+	req = shared__cluster_event_req__unpack(NULL, call->body.len,
+						call->body.data);
+	assert_non_null(req);
+
+	assert_string_equal(req->event->hostname, dss_hostname);
+	assert_int_equal((uint32_t)req->event->rank, (uint32_t)*rank);
+	assert_int_equal((uint32_t)req->event->id, (uint32_t)id);
+	assert_string_equal(req->event->msg, msg);
+	assert_int_equal((uint32_t)req->event->type, (uint32_t)type);
+	assert_int_equal((uint32_t)req->event->severity, (uint32_t)sev);
+	assert_string_equal(req->event->hw_id, hid);
+	assert_string_equal(req->event->job_id, jid);
+	assert_int_equal(uuid_parse(req->event->pool_uuid, req_puuid), 0);
+	assert_int_equal(uuid_compare(req_puuid, *puuid), 0);
+	assert_int_equal(uuid_parse(req->event->cont_uuid, req_cuuid), 0);
+	assert_int_equal(uuid_compare(req_cuuid, *cuuid), 0);
+	assert_string_equal(req->event->obj_id, "1.1");
+	assert_string_equal(req->event->ctl_op, cop);
+	assert_string_equal(req->event->str_info, data);
+
+	/* Cleanup */
+	shared__cluster_event_req__free_unpacked(req, NULL);
+	drpc__call__free_unpacked(call, NULL);
+}
+
+static void
+test_drpc_verify_notify_ras(void **state)
+{
+	uuid_t		puuid, cuuid;
+	d_rank_t	rank = 1;
+	daos_obj_id_t	oid = { .hi = 1, .lo = 1 };
+
+	mock_valid_drpc_resp_in_recvmsg(DRPC__STATUS__SUCCESS);
+	assert_int_equal(drpc_init(), 0);
+
+	assert_int_equal(uuid_parse("11111111-1111-1111-1111-111111111111",
+				    puuid), 0);
+	assert_int_equal(uuid_parse("22222222-2222-2222-2222-222222222222",
+				    cuuid), 0);
+
+	ds_notify_ras_event(RAS_RANK_NO_RESPONSE, "no response", RAS_TYPE_INFO,
+			    RAS_SEV_WARN, "exhwid", &rank, "exjobid", &puuid,
+			    &cuuid, &oid, "exctlop",
+			    "{\"people\":[\"bill\",\"steve\",\"bob\"]}");
+	verify_notify_ras(RAS_RANK_NO_RESPONSE, "no response", RAS_TYPE_INFO,
+			  RAS_SEV_WARN, "exhwid", &rank, "exjobid", &puuid,
+			  &cuuid, &oid, "exctlop",
+			  "{\"people\":[\"bill\",\"steve\",\"bob\"]}");
+
+	drpc_fini();
+}
+
+static void
+verify_notify_ras_min_viable(ras_event_t id, char *msg, ras_type_t type,
+			     ras_sev_t sev)
+{
+	Drpc__Call		*call;
+	Shared__ClusterEventReq	*req;
+
+	call = drpc__call__unpack(NULL, sendmsg_msg_iov_len,
```

Comment:
possible to call verify_notify_ras with the desired params? to avoid code duplication

Reply:
I could do but I would need to add null checks into verify_notify_ras and I'm not sure it's worthwhile adding that logic, I could change the verify functions to take the expected strings and integers rather than translating the types inside the function but I'd like to do that as a subsequent piece of work maybe when we add more events.

Reply:
will look into this as part of https://jira.hpdd.intel.com/browse/DAOS-6436 if that's okay